graphql_compiler/compiler/expressions.py

```diff
@@ -885,6 +949,47 @@ def to_gremlin(self):
                                   left=self.left.to_gremlin(),
                                   right=self.right.to_gremlin())
 
+    def to_cypher(self):
+        """Return a unicode object with the Cypher representation of this expression."""
+        self.validate()
+
+        # The Cypher versions of some operators require an inverted order of arguments.
+        regular_operator_format = u'({left} {operator} {right})'
+        inverted_operator_format = u'({right} {operator} {left})'
+        negated_inverted_operator_format = u'(NOT ({right} {operator} {left}))'
+        intersects_operator_format = u'any(_ {operator} {left} WHERE _ {operator} {right})'
+
+        # Null literals use 'is/is not' as (in)equality operators, while other values use '=/<>'.
+        if self.left == NullLiteral or self.right == NullLiteral:
+            translation_table = {
+                u'=': (u'IS', regular_operator_format),
+                u'!=': (u'IS NOT', regular_operator_format),
+            }
+        else:
+            translation_table = {
+                u'=': (u'=', regular_operator_format),
+                u'!=': (u'!=', regular_operator_format),
+                u'>=': (u'>=', regular_operator_format),
+                u'<=': (u'<=', regular_operator_format),
+                u'>': (u'>', regular_operator_format),
+                u'<': (u'<', regular_operator_format),
+                u'||': (u'OR', regular_operator_format),
+                u'&&': (u'AND', regular_operator_format),
+                u'contains': (u'IN', inverted_operator_format),
```

Comment:
Seems like contains and not_contains (as in does the list L contain item I) aren't [valid operators](https://neo4j.com/docs/cypher-manual/current/syntax/operators/#query-operators-comparison) in Cypher, but it's valid for searching in strings.

If we remove both of them, we can also take out the negated_inverted_operator format.

Reply:
`contains` and `not_contains` are what we call the operators, not what Cypher calls them. If you trace through the code, you'll see that we actually use Cypher's `IN` operator to emit the expressions that use `contains` and `not_contains`.

Reply:
Ah, gotcha-- my initial reading of the docs suggested that `IN` is just for searching for a substring in a string but I just tried it running a query to search in a list and it works, so my initial reading was mistaken.